src/lib/core/projection/projection.ts

```diff
@@ -0,0 +1,56 @@
+import {Injectable, Directive, ModuleWithProviders, NgModule, ElementRef} from '@angular/core';
+
+
+@Directive({
+  selector: 'md-host'
+})
+export class MdProjectionHostDirective {
+  constructor(public ref: ElementRef) {}
+}
+
+
+@Injectable()
+export class MdProjectionService {
+  private _replaceWith(toReplaceEl: HTMLElement, otherEl: HTMLElement) {
```

Comment:
We cannot replace all nodes by all nodes in the tree. Element is semantically accurate.